--- gnuplot-orig/term/cairo.trm	2007-09-13 23:12:53.000000000 -0700
+++ gnuplot/term/cairo.trm	2007-09-15 17:29:59.000000000 -0700
@@ -315,9 +315,9 @@
 	if ( plot.cr )
 		cairo_destroy(plot.cr);
 
-	surface = cairo_pdf_surface_create   (outstr /* output file name */,
-                                             plot.device_xmax /*double width_in_points*/,
-                                             plot.device_ymax /*double height_in_points*/);
+	surface = cairo_image_surface_create  (CAIRO_FORMAT_ARGB32,
+                                              plot.device_xmax /*double width_in_points*/,
+                                              plot.device_ymax /*double height_in_points*/);
 	plot.cr = cairo_create(surface);
 	cairo_surface_destroy( surface );
 
@@ -385,6 +385,7 @@
 void cairopdf_reset()
 {
 	/* finish the document */
+        cairo_surface_write_to_png(cairo_get_target(plot.cr), outstr);
 	cairo_destroy(plot.cr);
 
 	FPRINTF((stderr,"cairopdf_reset\n"));
